src/server/scripts/EasternKingdoms/zone_searing_gorge.cpp

```diff
@@ -0,0 +1,126 @@
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+* more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "ScriptMgr.h"
+#include "GameObject.h"
+#include "GameObjectAI.h"
+
+#include <mutex>
+
+enum PillowFight
+{
+    NPC_SLEEPY_DARK_IRON_WORKER = 14635,
+    NPC_CHAMBERMAID_PILLACLENCHER = 14636
+};
+
+class event_dark_iron_rush : public WorldMapScript
+{
+public:
+    event_dark_iron_rush() : WorldMapScript("event_dark_iron_rush", 0), duration(0), numWave(0), running(false) {}
+
+    void OnUpdate(Map* map, uint32 diff) override
+    {
+        if (running && numWave < 3)
+        {
+            duration += diff;
+            if (numWave == 0 && duration > 60000)
+            {
+                for (size_t i = 0; i < 10; i++)
+                {
+                    Position pos = { -6568.03f + frand(-5,5),-1306.73f + frand(-5,5),208.7f, frand(0.0f, 2.0f * (float)M_PI) };
+                    TempSummon* summon = map->SummonCreature(NPC_SLEEPY_DARK_IRON_WORKER, pos, nullptr, 60000);
+                    summon->SetTempSummonType(TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT);
+
+                    if(i == 0)
+                        summon->ToCreature()->AI()->Talk(0);
+                }
+                numWave++;
+            }
+            else if (numWave == 1 && duration > 180000)
+            {
+                for (size_t i = 0; i < 10; i++)
+                {
+                    Position pos = {-6483.42f + frand(-5,5), -1350.1f + frand(-5,5), 212.1f, frand(0.0f, 2.0f * (float)M_PI) };
+                    TempSummon* summon = map->SummonCreature(NPC_SLEEPY_DARK_IRON_WORKER, pos, nullptr, 60000);
+                    summon->SetTempSummonType(TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT);
+
+                    if (i == 0)
+                        summon->ToCreature()->AI()->Talk(1);
+                }
+                numWave++;
+            }
+            else if (numWave == 2 && duration > 300000)
+            {
+                Position pos = { -6545.88f, -1345.29f, 208.89f };
+                TempSummon* summon = map->SummonCreature(NPC_CHAMBERMAID_PILLACLENCHER, pos, nullptr, 300000);
+                summon->SetTempSummonType(TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT);
+                numWave++;
+                running = false;
+            }
+        }
+    }
+
+    void Trigger()
+    {
+        std::lock_guard<std::mutex> triggerGuard(triggerLock);
+        if (!running)
+        {
+            running = true;
+            duration = 0;
+            numWave = 0;
+        }
+    }
+
+    uint32_t duration;
+    uint32_t numWave;
+    bool running;
+
+    std::mutex triggerLock;
+};
+
+static event_dark_iron_rush *edir_p = NULL;
```

Comment:
probably there is a spell that launches the event, maybe its server-side

Reply:
https://wotlk.evowow.com/?object=179829
???

Reply:
(for those of you using ad block software : )
https://woehead.way-of-elendil.fr/?object=179829
(Dark Iron Pillow Event Generator)